python-package/lightgbm/sklearn.py

```diff
@@ -507,6 +507,9 @@ def __init__(
         random_state: Optional[Union[int, np.random.RandomState, np.random.Generator]] = None,
         n_jobs: Optional[int] = None,
         importance_type: str = "split",
+        *,
+        early_stopping: Union[bool, int] = False,
+        validation_fraction: float = 0.1,
```

Comment:
@jameslamb, I have removed the option of having `validation_fraction=None` and performing early stopping on the train set.

Early stopping in lightgbm is disabled when the validation set equals the train set and I did not want to alter that behavior in this PR, so I have decided to deviate a little from the scikit-learn implementation of auto early stopping and to not allow `validation_fraction=None`. 

Let me know your thoughts on it.